plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestDecimalCoercers.java

```diff
@@ -83,4 +89,62 @@ private void assertDecimalToIntCoercion(Type fromType, Object valueToBeCoerced,
         assertThat(blockToNativeValue(toType, coercedValue))
                 .isEqualTo(expectedValue);
     }
+
+    @Test
+    public void testTinyintToDecimalCoercion()
+    {
+        for (DecimalType decimalType : List.of(createDecimalType(), createDecimalType(30, 20))) {
+            assertDecimalCoercion(TINYINT, 0L, decimalType, Int128.ZERO);
+            assertDecimalCoercion(TINYINT, 0L, decimalType, Int128.ZERO);
+            assertDecimalCoercion(TINYINT, 12L, decimalType, Int128.valueOf(12));
+            assertDecimalCoercion(TINYINT, -12L, decimalType, Int128.valueOf(-12));
+            assertDecimalCoercion(TINYINT, (long) Byte.MAX_VALUE, decimalType, Int128.valueOf(127));
+            assertDecimalCoercion(TINYINT, (long) Byte.MAX_VALUE, decimalType, Int128.valueOf(127));
```

Comment:
This is a duplicate. Also for all of these, you should be able to use the max value constant for the expected value.  Something like this:
```suggestion
            assertDecimalCoercion(TINYINT, (long) Byte.MAX_VALUE, decimalType, Int128.valueOf(Byte.MAX_VALUE));
```